lib/initialize.ts

```diff
@@ -34,26 +36,33 @@ In order for the ui-extension-sdk to function correctly, your app needs to be ru
 Learn more about local development with the ui-extension-sdk here:
   https://www.contentful.com/developers/docs/extensibility/ui-extensions/faq/#how-can-i-use-the-ui-extension-sdk-locally`)
     }
-    connectDeferred.promise.then(
-      ([channel, params, messageQueue]: [Channel, ConnectMessage, unknown[]]) => {
-        const api = apiCreator(channel, params, currentWindow)
 
-        let customApi
-        if (typeof makeCustomApi === 'function') {
-          customApi = makeCustomApi(channel, params)
-        }
+    if (!initializedSdks) {
+      initializedSdks = new Promise((resolve) => {
+        connectDeferred.promise.then(([channel, params, messageQueue]) => {
+          const api = apiCreator(channel, params, currentWindow)
 
-        // Handle pending incoming messages.
-        // APIs are created before so handlers are already
-        // registered on the channel.
-        messageQueue.forEach((m) => {
-          // TODO Expose private handleMessage method
-          ;(channel as any)._handleMessage(m)
+          let customApi
+          if (typeof makeCustomApi === 'function') {
+            customApi = makeCustomApi(channel, params)
+          }
+
+          // Handle pending incoming messages.
+          // APIs are created before so handlers are already
+          // registered on the channel.
+          messageQueue.forEach((m) => {
+            // TODO Expose private handleMessage method
+            ;(channel as any)._handleMessage(m)
+          })
+
+          resolve([api, customApi])
```

Comment:
I think if you just return the value in the `then`, you don't need to create wrap it with a new promise and should get the promise automatically

Reply:
Good point. Should make the code less indented